packages/auth/amplify_auth_cognito_test/test/plugin/fetch_current_device_test.dart

```diff
@@ -0,0 +1,99 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import 'package:amplify_auth_cognito_dart/amplify_auth_cognito_dart.dart';
+import 'package:amplify_auth_cognito_dart/src/credentials/cognito_keys.dart';
+import 'package:amplify_auth_cognito_dart/src/credentials/device_metadata_repository.dart';
+import 'package:amplify_auth_cognito_dart/src/sdk/cognito_identity_provider.dart';
+import 'package:amplify_auth_cognito_test/common/mock_clients.dart';
+import 'package:amplify_auth_cognito_test/common/mock_config.dart';
+import 'package:amplify_auth_cognito_test/common/mock_secure_storage.dart';
+import 'package:amplify_core/amplify_core.dart';
+import 'package:test/test.dart';
+
+void main() {
+  AmplifyLogger().logLevel = LogLevel.verbose;
+
+  final userPoolKeys = CognitoUserPoolKeys(userPoolConfig);
+  final identityPoolKeys = CognitoIdentityPoolKeys(identityPoolConfig);
+  final testAuthRepo = AmplifyAuthProviderRepository();
+  final mockDevice = DeviceType(deviceKey: deviceKey);
+  final mockDeviceResponse = GetDeviceResponse(device: mockDevice);
+
+  late DeviceMetadataRepository repo;
+  late AmplifyAuthCognitoDart plugin;
+
+  group('fetchCurrentDevice', () {
+    setUp(() async {
+      final secureStorage = MockSecureStorage();
+      seedStorage(
+        secureStorage,
+        userPoolKeys: userPoolKeys,
+        identityPoolKeys: identityPoolKeys,
+        deviceKeys: CognitoDeviceKeys(userPoolConfig, username),
+      );
+      plugin = AmplifyAuthCognitoDart(
+        secureStorageFactory: (_) => secureStorage,
+      );
+      await plugin.configure(
+        config: mockConfig,
+        authProviderRepo: testAuthRepo,
+      );
+      repo = plugin.stateMachine.getOrCreate<DeviceMetadataRepository>();
+    });
+
+    group('Cognito GetDevice returns successfully.', () {
+      setUp(() async {
+        final mockIdp = MockCognitoIdentityProviderClient(
+          getDevice: () async => mockDeviceResponse,
+          forgetDevice: () async {},
+        );
+        plugin.stateMachine.addInstance<CognitoIdentityProviderClient>(mockIdp);
+      });
+
+      test(
+          'This test should get the current device, where the current device id is equal to the local device id',
+          () async {
+        final secrets = await repo.get(username);
+        final currentDeviceKey = secrets?.deviceKey;
+        expect(currentDeviceKey, isNotNull);
+        final currentDevice = await plugin.fetchCurrentDevice();
+        expect(currentDeviceKey, currentDevice.id);
+      });
+
+      test(
+          'This test should throw a DeviceNotTrackedException when current device key is null',
```

Comment:
I think this test is also under the wrong group. Move it to the `should throw` group. 

```suggestion
          'a DeviceNotTrackedException when current device key is null',
```